### PR TITLE
Propose some potential learning objectives

### PR DESCRIPTION
**Pull request by robmoss**
This proposes several learning objectives for the "Effective use of git" and "Collaborating" sections.

We should probably aim to have an exercise for each objective, so that the reader can test their knowledge.

### CODE REVIEW

**Comment by robmoss** [comment]
I've rebased this branch so that it builds successfully with mdbook-admonish 1.6.0. We can now use tabbed code blocks, which might be useful for exercises that accompany the learning objectives.

**Comment by robmoss** [comment]
I've added some exercises at the end of the chapter, which should test whether the participants are able to apply all of the skills we aim to cover in this section. The skills, in turn, should be sufficient for the participants to conduct peer code review.

- Do you think the exercises are sufficiently clear and well-structured/ordered?
- Do you think there's anything missing from these exercises or the learning objectives?

**Comment by robmoss** [comment]
Note to self: also add some questions at the end of the "Version control concepts" section, such as:

- Do you have an existing project or piece of code that could be stored in a repository?
- Have you recently written any code that could be recorded as one or more commits?
- Have you ever written some exploratory code or analysis that could be stored in a separate branch?

**Comment by jomonman537** [comment]
I had a read of the list of objectives. Overall, the objectives look pretty good and cover most of what the workbook goes through. I have two main comments to add regarding the list of objectives:

1) In the first group of objectives, there is nothing in chapter 1 that talks about identifying projects and tasks where version control would be appropriate. I would suggest adding a section in chapter 1 that talks about this explicitly.

2) In the third group of objectives, I would add a new objective "understand how to conduct peer code review respectfully". Given the discussions we had in today's meeting, I think it would be appropriate to make this objective explicit in the list of objectives.

**Comment by robmoss** [comment]
Thanks very much for these suggestions, James.

>     1. In the first group of objectives, there is nothing in chapter 1 that talks about identifying projects and tasks where version control would be appropriate. I would suggest adding a section in chapter 1 that talks about this explicitly.

I suspect it will be a real challenge to come up anything except extremely vague statements. I'll have to give this some thought.

**Comment by robmoss** [comment]
> 

@jomonman537 on the [What is version control?](https://robmoss.github.io/git-is-my-lab-book/version-control/what-is-version-control.html) page, we identify some key advantages of using version control. Does this go some way towards identifying projects and tasks where version control would be appropriate? If so, we could add a section before the exercises at the end of this chapter in which we reiterate these applications, and add any other applications that come to mind.

**Comment by jomonman537** [comment]
I think listing the key uses of version control in section 1.1 should be fine. We could add a question in the chapter 1 exercises which would prompt people to think about how they can use version control in their research project. For example:

"In this chapter, we outlined that version control can be used to... [summarise the applications]. How can you use version control in your research project? How would that benefit you?"

This would prompt people to reflect on what they have learnt from the chapter and apply them to their work. Both of these actions would enhance the retention of what they learn from the chapter.

**Comment by robmoss** [comment]
> I think listing the key uses of version control in section 1.1 should be fine. We could add a question in the chapter 1 exercises which would prompt people to think about how they can use version control in their research project. For example:
> 
> "In this chapter, we outlined that version control can be used to... [summarise the applications]. How can you use version control in your research project? How would that benefit you?"
> 
> This would prompt people to reflect on what they have learnt from the chapter and apply them to their work. Both of these actions would enhance the retention of what they learn from the chapter.

Thanks very much James, this is helpful. I've [updated the exercises](https://github.com/robmoss/git-is-my-lab-book/blob/9e2a66658ad81f32a14af7387dc35c0a3bd344d4/src/version-control/exercise-using-version-control.md) so that it now begins with a reminder of some ways in which version control can support research activities, and then added a set of questions that ask whether they've had any experiences in the past where version control would have been useful. What do you think?

**Comment by jomonman537** [comment]
Those exercises are very helpful. It is actually good that readers will be asked to reflect from the past on how version control could have been used to assist them in their work and think about how they will use version control in the future. I feel that doing both should solidify the value of version control for the reader.

**Comment by robmoss** [comment]
> Those exercises are very helpful. It is actually good that readers will be asked to reflect from the past on how version control could have been used to assist them in their work and think about how they will use version control in the future. I feel that doing both should solidify the value of version control for the reader.

Sorry, I sent you an email after pushing the commit, so I'll copy the relevant text here:

"Hopefully even if they don't have any past challenges, the examples I've listed will prompt them to think about how version control could benefit them. Do you think we should also explicitly ask them the broader questions in your comment — How can you use version control in your research project? How would that benefit you? — and, if so, do you think they should be the first questions we ask?"

**Comment by jomonman537** [comment]
Maybe putting the broader questions in the exercise would help readers understand the purpose of the exercise. The first question (how can you use version control in your research project) could be placed above the sub-list of questions while the second question (how would version control benefit you) could be placed below the sub-list of questions.

**Comment by robmoss** [comment]
> Maybe putting the broader questions in the exercise would help readers understand the purpose of the exercise. The first question (how can you use version control in your research project) could be placed above the sub-list of questions while the second question (how would version control benefit you) could be placed below the sub-list of questions.

To make sure I understand, this would look like:

1. How can you use version control in your research project(s)?
2. Have you experienced any issues or challenges in your career where version control would have been helpful? For example:
   - ...
   - ...
3. We'd also like you to consider how version control might be applied to your current work:
   - ...
   - ...
4. How would using version control benefit you?

**Comment by robmoss** [comment]
Now that I see it written out, I feel there's a fair amount of overlap between Q1 and Q3.

**Comment by jomonman537** [comment]
Oh, I meant putting the additional questions around the second set of dotpoints, not around both the first and second sets. Here is where I would place the questions.

We'd also like you to consider how version control might be applied to your current work. **How can you use version control in your research project(s)?**

- Do you have an existing project or piece of code that could benefit from being stored in a repository?
- Have you recently written any code that could be recorded as one or more commits?
- If so, what would you write for the commit messages?
- Have you written some exploratory code or analysis that could be stored in a separate branch?

**Having looked at the use of version control in the past and present, how would using version control benefit you?**

**Comment by robmoss** [comment]
Okay, that make sense! Thank you, I'm glad I asked :)

**Comment by robmoss** [comment]
Hopefully this [revised version](https://github.com/robmoss/git-is-my-lab-book/blob/cac2aa422687bbcbcd576f85e84bb1641f4d4b11/src/version-control/exercise-using-version-control.md) looks good to everyone, and we can merge this branch!

**Comment by jomonman537** [comment]
That looks pretty good. Happy for you to merge this branch.

**Comment by robmoss** [comment]
> That looks pretty good. Happy for you to merge this branch.

Thanks for all of your input and help, James!

 @EamonConway are you happy?

File: src/learning-objectives.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Learning objectives
+
+This page defines the learning objectives for individual sections.
+These are skills that the reader should be able to demonstrate after reading through the relevant section, and completing any exercises in that section.
+
+## Version control concepts
+
+After completing [this section](version-control/), you should be able to identify how to apply version control concepts to some of your existing work.
```
Comment (EamonConway) [inline src/learning-objectives.md]:
```suggestion
After completing [this section](version-control/), you should be able to identify how to apply version control concepts to your existing work.
```

File: src/learning-objectives.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+This includes be able to:
```
Comment (EamonConway) [inline src/learning-objectives.md]:
```suggestion
This includes being able to:
```

File: src/using-git/exercise-use-a-remote-repository.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Exercise: use a remote repository
+
+In this exercise, you will work use a remote repository to synchronise and merge changes between multiple local repositories, starting from the local `git-exercise` repository that you created in [the previous exercise](exercise-create-a-local-repository.md).
```
Comment (EamonConway) [inline src/using-git/exercise-use-a-remote-repository.md]:
```suggestion
In this exercise, you will use a remote repository to synchronise and merge changes between multiple local repositories, starting from the local `git-exercise` repository that you created in [the previous exercise](exercise-create-a-local-repository.md).
```

File: src/using-git/exercise-use-a-remote-repository.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+   - A table of the summary statistics printed by the plotting scripting (see the [Markdown Guide](https://www.markdownguide.org/extended-syntax/#tables)); and
+   - The figure produced by the plotting script (see the [Markdown Guide](https://www.markdownguide.org/basic-syntax/#images-1)).
+
+   Record your changes in a new commit.
+
+2. Push this new branch to the remote repository.
+
+## Merge the report into the original repository
+
+1. In your original `git-exercise` repository, check out the `feature/report` branch from the remote repository and verify that it now contains the file `report.md`.
```
Comment (EamonConway) [inline src/using-git/exercise-use-a-remote-repository.md]:
```suggestion
1. In your original `git-exercise` repository, checkout the `feature/report` branch from the remote repository and verify that it now contains the file `report.md`.
```

File: src/using-git/exercise-use-a-remote-repository.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+2. Merge this branch into your `main` or `master` branch, and add a new tag `milestone-3-report`.
+
+3. Push the updated `main` or `master` branch to the remote repository.
+
+## Update the new local repository
+
+1. In your `git-exercise-2` repository, check out the `main` or `master` branch and pull changes from the remote repository.
```
Comment (EamonConway) [inline src/using-git/exercise-use-a-remote-repository.md]:
```suggestion
1. In your `git-exercise-2` repository, checkout the `main` or `master` branch and pull changes from the remote repository.
```

**Review by EamonConway — APPROVED** [review]
I've made a couple of suggestions on spelling/grammar for review. 
Once done, I think the branch is suitable for merging.

**Comment by robmoss** [comment]
Thank you both!

**Pull request merged**